Dinah.Core/_IO/JsonFilePersister.cs

```diff
@@ -44,6 +43,7 @@ protected JsonFilePersister(T target, string path, string? jsonPath = null)
 			validatePath(path);
 
 			Path = path;
+			_locks.TryAdd(Path, new());
```

Comment:
You need to make sure the other constructor also calls TryAdd

Reply:
Oh dead god -- what made me think I could do this on 4 1/2 hours of sleep? I can't believe I missed the other ctor. Fixed